src/xref2/env.ml

```diff
@@ -655,22 +660,28 @@ let initial_env :
  fun t resolver ->
   let open Odoc_model.Lang.Compilation_unit in
   let initial_env =
-    let m = module_of_unit t in
+    let m = Component.Delayed.put (fun () -> module_of_unit t) in
     empty |> add_module (t.id :> Identifier.Path.Module.t) m
   in
   let initial_env = set_resolver initial_env resolver in
   List.fold_right
     (fun import (imports, env) ->
       match import with
-      | Import.Resolved root ->
-          let unit = resolver.resolve_unit root in
-          let m = module_of_unit unit in
-          let env = add_module (unit.id :> Identifier.Path.Module.t) m env in
+      | Import.Resolved (root, name) ->
+          let id = `Root(root, name) in
+          let m =
+            Component.Delayed.put (fun () ->
+              let unit = resolver.resolve_unit root in
+              module_of_unit unit)
+          in
+          let env = add_module id m env in
```

Comment:
I'm not convinced this does anything useful right now aside from fill in the cache in `compilation_unit.ml` - I think root module lookup doesn't look in `env.elts` - @Julow would know for sure.

Reply:
(There's a bit of an impedance mismatch here as `Env.t` is an immutable object, but the root module lookup is done on demand, which is why it's a bit odd)

Reply:
I think we can remove this, the speed improvements is around 3%.
I tried to remove that code entirely (and the `imports` value) and got no diffs on Core's output. It seems this value is only used by the `depends` command ?

Reply:
We certainly need to turn all the unresolved imports into resolved ones (or forward) for the `odoc html-deps` to work.